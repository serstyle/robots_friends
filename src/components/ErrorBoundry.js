import React, { Component } from 'react'

class ErrorBoundary extends Component {
  constructor (props) {
    super(props)
    this.state = { hasError: false }
  }

  componentDidCatch (error, info) {
    this.setState({ hasError: true })
  }

  somethingWentWrong = (hasError, children) =>{
    if (hasError) {
      return <h1>Something went wrong.</h1>
    }
    return children
  }
  

  render () {
   return this.somethingWentWrong(this.state.hasError, this.props.children)
}
}
export default ErrorBoundary