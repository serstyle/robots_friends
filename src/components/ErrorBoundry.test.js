import React from 'react';
import { shallow, mount, render } from 'enzyme';

import ErrorBoundry from './ErrorBoundry'

const wrapper = shallow( <ErrorBoundry > </ErrorBoundry> );

it('Test ErrorBoundry', () => {
	expect(wrapper).toMatchSnapshot()
})


it('TEST somethingWentWrong function true', ()=> {

	expect(wrapper.instance().somethingWentWrong(true)).toEqual(<h1>Something went wrong.</h1>)
})



it('TEST somethingWentWrong function false', ()=> {
	const children = '<Something />'
	expect(wrapper.instance().somethingWentWrong(false, children)).toEqual('<Something />')
})

it('test error', ()=> {
	wrapper.instance().componentDidCatch()
	expect(wrapper.state().hasError).toEqual(true)
})

