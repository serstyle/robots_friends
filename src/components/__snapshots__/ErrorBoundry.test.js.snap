// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ErrorBoundry 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorBoundary>
     
</ErrorBoundary>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): " ",
  Symbol(enzyme.__nodes__): Array [
    " ",
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
